fhir-server/src/main/java/org/linuxforhealth/fhir/server/util/RestAuditLogger.java

```diff
@@ -746,6 +774,142 @@ protected static AuditLogEntry populateAuditLogEntry(AuditLogEntry entry, HttpSe
+            if (resourceContext.getResourceType() != null) {
+                entry.getContext().getData().setResourceType(resourceContext.getResourceType());
+            }
+            if (resourceContext.getId() != null) {
+                entry.getContext().getData().setId(resourceContext.getId());
+            }
+            if (resourceContext.getVersionId() != null) {
+                entry.getContext().getData().setVersionId(resourceContext.getVersionId());
+            }
+        }
+        return entry;
+    }
+
+
+    /**
+     * Populates the passed audit log entry for Bundle entry.
+     * This method will populate the resource id, resource type and version id from the
+     * Bundle response entry location when the return preference is "OperationOutcome".
+     * When the return preference is "representation" the populateAuditLogEntry method will
+     * populate the resource id, resource type and version id.
+     * @param entry
+     *  The AuditLogEntry to be populated.
+     * @param responseEntry
+     *  The Bundle response entry.
+     * @param request
+     *  The HttpServletRequest representation of the REST request.
+     * @param resource
+     *  The Resource object.
+     * @param startTime
+     *  The start time of the request execution.
+     * @param endTime
+     *  The end time of the request execution.
+     * @param responseStatus
+     *  The response status.
+     * @return AuditLogEntry - an initialized audit log entry.
+     */
+    protected static AuditLogEntry populateBundleAuditLogEntry(AuditLogEntry entry, Bundle.Entry responseEntry, HttpServletRequest request, Resource resource,
+            Date startTime, Date endTime, Response.Status responseStatus) { 
+        final String METHODNAME = "populateBundleAuditLogEntry";
+        log.entering(CLASSNAME, METHODNAME);
+
+        // call populateAuditLogEntry to populate common attributes to all rest
+        populateAuditLogEntry(entry, request, resource, startTime, endTime, responseStatus);
+        if (HTTPReturnPreference.REPRESENTATION.equals(FHIRRequestContext.get().getReturnPreference())) {
+            return entry;
+        }
+        if (responseEntry.getResponse() == null || responseEntry.getResponse().getLocation() == null) {
+            return entry;
+        }
+        // Populate the resource id, resource type and version id from the
+        // Bundle response entry location when the return preference is "OperationOutcome".
+        String location = responseEntry.getResponse().getLocation().getValue();
+        String[] parts = location.split("/");
+        if (parts.length > 3) {
+            Collections.reverse(Arrays.asList(parts));
+            entry.getContext().setData(Data.builder().build());
+            entry.getContext().getData().setResourceType(parts[3]);
+            entry.getContext().getData().setVersionId(parts[0]);
+            entry.getContext().getData().setId(parts[2]);
```

Comment:
is it possible to always populate the AuditLogEntry in the same way (e.g. from the response location)?
do we not set that unless the return preference is OperationOutcome?